integration_tests/validator_out_test.go

```diff
@@ -0,0 +1,149 @@
+		s.Require().NoError(err)
+		err = s.sendToCosmos(valAddress, sdk.NewInt(200))
+		s.Require().NoError(err, "error sending test denom to cosmos")
+		kb, err := val.keyring()
+		s.Require().NoError(err)
+		clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", valAddress)
+		s.Require().NoError(err)
+		gbQueryClient := types.NewQueryClient(clientCtx)
+		bankQueryClient := banktypes.NewQueryClient(clientCtx)
+		var gravityDenom string
+		s.Require().Eventuallyf(func() bool {
+			res, err := bankQueryClient.AllBalances(context.Background(),
+				&banktypes.QueryAllBalancesRequest{
+					Address: valAddress.String(),
+				})
+			if err != nil {
+				return false
+			}
+			denomRes, err := gbQueryClient.ERC20ToDenom(context.Background(),
+				&types.ERC20ToDenomRequest{
+					Erc20: testERC20contract.String(),
+				})
+			if err != nil {
+				s.T().Logf("error querying ERC20 denom %s, %e", testERC20contract.String(), err)
+				return false
+			}
+			s.Require().False(denomRes.CosmosOriginated, "ERC20-originated denom marked as cosmos originated")
+			gravityDenom = denomRes.Denom
+
+			for _, coin := range res.Balances {
+				if coin.Denom == gravityDenom && coin.Amount.Equal(sdk.NewInt(200)) {
+					return true
+				}
+			}
+
+			s.T().Logf("balance not found, received %v", res.Balances)
+
+			return false
+		}, 105*time.Second, 10*time.Second, "balance never found on cosmos")
+
+		s.T().Logf("sending to ethereum")
+		sendToEthereumMsg := types.NewMsgSendToEthereum(
+			valAddress,
+			s.chain.validators[0].ethereumKey.address,
+			sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(100)},
+			sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(1)},
+		)
+		// Send NewMsgSendToEthereum Message
+		response, err := s.chain.sendMsgs(*clientCtx, sendToEthereumMsg)
+		s.Require().NoError(err)
+		s.Require().Equal(uint32(0), response.Code)
+
+		// Confirm batchtx
+		queryClient := types.NewQueryClient(clientCtx)
+		s.Require().Eventuallyf(func() bool {
+			res, err := queryClient.BatchTxConfirmations(context.Background(), &types.BatchTxConfirmationsRequest{BatchNonce: 1, TokenContract: testERC20contract.String()})
+			if err != nil {
+				s.T().Logf("error: %s", err)
+			}
+			if len(res.GetSignatures()) != 0 {
+				return true
+			}
+			return false
+		}, 5*time.Minute, 10*time.Second, "Can't find Batchtx signing info")
+
+		// Check jail status of validators
+		s.Require().Eventuallyf(func() bool {
+			orch3Key := s.chain.validators[3]
+			keyring3, err := orch3Key.keyring()
+			s.Require().NoError(err)
+			val3Address, err := s.chain.validators[3].keyInfo.GetAddress()
+			s.Require().NoError(err)
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &keyring3, "val", val3Address)
+			s.Require().NoError(err)
+			newQ := stakingtypes.NewQueryClient(clientCtx)
+			valThree, err := newQ.Validator(context.Background(), &stakingtypes.QueryValidatorRequest{ValidatorAddr: sdk.ValAddress(val3Address).String()})
```

Comment:
this will spin up a new validator node under the same address, hence it'll double-sign / be jailed?

Reply:
actually the first line of the test will stop the orchestrator3.

the validator node 3 should still be available, we are simply checking that it should be jailed because of missing signature from the batch constructed previously.

I will add more comment in the test scenario

Reply:
oh ok https://github.com/crypto-org-chain/gravity-bridge/pull/161/files#diff-a24fb5338db7310da3370a6c6257dbce3de1cbe763968a8ec0da838ed4daa7f9R20

Reply:
validator3 would be jailed for not signing the gravity-bridge cross chain messages?

Reply:
yes, for the test, the SignedBatchesWindow has been set to 15 secs